vector_test.go

```diff
@@ -0,0 +1,403 @@
+	time3, _ := time.Parse("15:04:05.000", "10:31:45.987")
+
+	duration1 := Duration{0, 1, 1920000000000}
+	duration2 := Duration{1, 1, 1920000000000}
+	duration3 := Duration{31, 0, 60000000000}
+
+	map1 := make(map[string]int)
+	map1["a"] = 1
+	map1["b"] = 2
+	map1["c"] = 3
+	map2 := make(map[string]int)
+	map2["abc"] = 123
+	map3 := make(map[string]int)
+
+	tests := []struct {
+		name       string
+		cqlType    string
+		value      interface{}
+		comparator func(interface{}, interface{})
+	}{
+		{name: "ascii", cqlType: TypeAscii.String(), value: []string{"a", "1", "Z"}},
+		{name: "bigint", cqlType: TypeBigInt.String(), value: []int64{1, 2, 3}},
+		{name: "blob", cqlType: TypeBlob.String(), value: [][]byte{[]byte{1, 2, 3}, []byte{4, 5, 6, 7}, []byte{8, 9}}},
+		{name: "boolean", cqlType: TypeBoolean.String(), value: []bool{true, false, true}},
+		{name: "counter", cqlType: TypeCounter.String(), value: []int64{5, 6, 7}},
+		{name: "decimal", cqlType: TypeDecimal.String(), value: []inf.Dec{*inf.NewDec(1, 0), *inf.NewDec(2, 1), *inf.NewDec(-3, 2)}},
+		{name: "double", cqlType: TypeDouble.String(), value: []float64{0.1, -1.2, 3}},
+		{name: "float", cqlType: TypeFloat.String(), value: []float32{0.1, -1.2, 3}},
+		{name: "int", cqlType: TypeInt.String(), value: []int32{1, 2, 3}},
+		{name: "text", cqlType: TypeText.String(), value: []string{"a", "b", "c"}},
+		{name: "timestamp", cqlType: TypeTimestamp.String(), value: []time.Time{timestamp1, timestamp2, timestamp3}},
+		{name: "uuid", cqlType: TypeUUID.String(), value: []UUID{MustRandomUUID(), MustRandomUUID(), MustRandomUUID()}},
+		{name: "varchar", cqlType: TypeVarchar.String(), value: []string{"abc", "def", "ghi"}},
+		{name: "varint", cqlType: TypeVarint.String(), value: []uint64{uint64(1234), uint64(123498765), uint64(18446744073709551615)}},
+		{name: "timeuuid", cqlType: TypeTimeUUID.String(), value: []UUID{TimeUUID(), TimeUUID(), TimeUUID()}},
+		{
+			name:    "inet",
+			cqlType: TypeInet.String(),
+			value:   []net.IP{net.IPv4(127, 0, 0, 1), net.IPv4(192, 168, 1, 1), net.IPv4(8, 8, 8, 8)},
+			comparator: func(e interface{}, a interface{}) {
+				expected := e.([]net.IP)
+				actual := a.([]net.IP)
+				assertEqual(t, "vector size", len(expected), len(actual))
+				for i, _ := range expected {
+					// TODO(lantoniak): Find a better way to compare IP addresses
+					assertEqual(t, "vector", expected[i].String(), actual[i].String())
+				}
+			},
+		},
+		{name: "date", cqlType: TypeDate.String(), value: []time.Time{date1, date2, date3}},
+		{name: "time", cqlType: TypeTimestamp.String(), value: []time.Time{time1, time2, time3}},
+		{name: "smallint", cqlType: TypeSmallInt.String(), value: []int16{127, 256, -1234}},
+		{name: "tinyint", cqlType: TypeTinyInt.String(), value: []int8{127, 9, -123}},
+		{name: "duration", cqlType: TypeDuration.String(), value: []Duration{duration1, duration2, duration3}},
+		// TODO(lantonia): Test vector of custom types
```

Comment:
NIT: a couple of TODOs in this test still

Reply:
👍